api/v1alpha1/ionoscloudmachine_types_test.go

```diff
@@ -457,6 +457,34 @@ var _ = Describe("IonosCloudMachine Tests", func() {
 			Expect(k8sClient.Update(context.Background(), m)).ToNot(Succeed())
 		})
 	})
+	Context("LanID", func() {
+		It("should allow setting an existing LanID in the spec", func() {
+			m := defaultMachine()
+			m.Spec.LanID = ptr.To("1")
+			Expect(k8sClient.Create(context.Background(), m)).To(Succeed())
+			Expect(m.Spec.LanID).To(Equal(ptr.To("1")))
+		})
+		It("should allow setting null", func() {
+			m := defaultMachine()
+			Expect(k8sClient.Create(context.Background(), m)).To(Succeed())
+			Expect(m.Spec.LanID).To(BeNil())
+		})
+		It("should not allow setting empty LanID", func() {
+			m := defaultMachine()
+			m.Spec.LanID = ptr.To("")
+			Expect(k8sClient.Create(context.Background(), m)).ToNot(Succeed())
+		})
+		It("should be immutable", func() {
+			m := defaultMachine()
+			m.Spec.LanID = ptr.To("1")
+			Expect(k8sClient.Create(context.Background(), m)).To(Succeed())
+			Expect(m.Spec.LanID).To(Equal(ptr.To("1")))
+			m.Spec.LanID = ptr.To("2")
+			Expect(k8sClient.Update(context.Background(), m)).ToNot(Succeed())
+			m.Spec.LanID = ptr.To("")
+			Expect(k8sClient.Update(context.Background(), m)).ToNot(Succeed())
```

Comment:
Clearing the LAN ID by assigning `nil` still works.

Reply:
Sure, but it doesn't.

```go
m.Spec.NetworkID = nil
Expect(k8sClient.Update(context.Background(), m)).To(Succeed())
Expect(m.Spec.NetworkID).To(BeNil())
Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(m), m)).To(Succeed()) // even refresh it
Expect(m.Spec.NetworkID).To(BeNil())
```

This should fail if the immutability works. But it doesn't.